tests/migration_test.go

```diff
@@ -2882,53 +2882,29 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				Expect(supportedFeatures).Should(Not(ContainElement(false)),
 					"copy features must be supported on node")
 			}
-			getOtherNodes := func(nodeList *k8sv1.NodeList, node *k8sv1.Node) (others []*k8sv1.Node) {
-				for _, curNode := range nodeList.Items {
-					if curNode.Name != node.Name {
-						others = append(others, &curNode)
-					}
-				}
-				return others
-			}
-			isHeterogeneousCluster := func() bool {
-				nodes := libnode.GetAllSchedulableNodes(virtClient)
-				for _, node := range nodes.Items {
-					hostModel := getNodeHostModel(&node)
-					otherNodes := getOtherNodes(nodes, &node)
-
-					foundSupportedNode := false
-					foundUnsupportedNode := false
-					for _, otherNode := range otherNodes {
-						if isModelSupportedOnNode(otherNode, hostModel) {
-							foundSupportedNode = true
-						} else {
-							foundUnsupportedNode = true
-						}
-
-						if foundSupportedNode && foundUnsupportedNode {
-							return true
-						}
-					}
-				}
-
-				return false
-			}
 
 			It("[test_id:6981]should migrate only to nodes supporting right cpu model", func() {
-				if !isHeterogeneousCluster() {
-					log.Log.Warning("all nodes have the same CPU model. Therefore the test is a happy-path since " +
-						"VMIs with default CPU can be migrated to every other node")
+				sourceNode, targetNode, err := tests.GetValidSourceNodeAndTargetNodeForHostModelMigration(virtClient)
```

Comment:
It not a must, I see that the functions `GetValidSourceNodeAndTargetNodeForHostModelMigration` and `AffinityToMigrateFromSourceToTargetAndBack` are used only in `migration_test.go` , can you please move them to `migration.go` file? it will reduce the pressure from `utils.go`

Reply:
Done.